File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @param {number} k
     * @return {number}
     */
    kthSmallest(root, k) {
        return this.findOrder(root)[k - 1].val; 
    }

    findOrder(root) {
        if (root == null) return []; 
        return [...this.findOrder(root.left), root, ...this.findOrder(root.right)]
    }
}

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        #BFS, at each elvel, we need to check which is most right, so basically the latest element per level of the BFS
        result = []

        if not root:
            return []

        frontier = [root]

        while len(frontier) > 0:
            new_front = []
            current = frontier[0]
            for el in frontier:
                if el:
                    if el.left:
                        new_front.append(el.left)
                    if el.right:
                        new_front.append(el.right)
                    current = el.val

            result.append(current)
            frontier = new_front


        return result
        